cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -488,7 +518,9 @@ def _add_lambda_cleanup(self, policy_statements, build_tags):
                         service="iam",
                         resource="role",
                         region="",
-                        resource_name="{0}/{1}-{2}-*".format(RESOURCE_NAME_PREFIX, self.image_name, "InstanceRole"),
+                        resource_name="{0}/{1}-*".format(
+                            RESOURCE_NAME_PREFIX, "-".join([self.image_name, "InstanceRole"])[0:50]
```

Comment:
We should avoid this 0:50 everywhere and create a function/methdo like `_build_resource_name`